src/lib.rs

```diff
@@ -61,7 +61,8 @@ pub use flat::{FlatSamples};
 pub use traits::Primitive;
 
 // Opening and loading images
-pub use dynimage::{guess_format, load, load_from_memory, load_from_memory_with_format, open,
+pub use io::free_functions::{guess_format, load};
+pub use dynimage::{load_from_memory, load_from_memory_with_format, open,
```

Comment:
How would you feel about deprecating some of these functions? We wouldn't have to remove them any time soon, but it would push people towards using the new (cleaner) API.

Reply:
Depends on how sure we are that we want the new set of methods to behave the same, or at least very similar, in terms of format deduction. Also, consider that the `Writer` equivalents do not yet exist. I'd keep them around undeprecated until at least the next major release to evaluate if any problem arises.

We should at minimum add a note to the documentation, however.